drivers/fastboot/fb_mmc.c

```diff
@@ -29,6 +29,27 @@ static int part_get_info_by_name_or_alias(struct blk_desc *dev_desc,
 {
 	int ret;
 
+
+	/** check if the MMC is configured for boot partition r/w.
+	 *  If so, then we aren't reading the partition table, it's not
+	 *  stored in the boot partition only the bootloader is and
+	 *  instead we are in raw block programming mode of the HW
+	 *  bootpartition, most likely to program the bootloader. If we
+	 *  try to read the partition from emmc while the emmc HW boot
+	 *  partition is activit, it'll just fail. Not a big deal, but
+	 *  the failure is a timeout in SDHC, so it takes a couple
+	 *  seconds to fail. Checking for that here prevents the query
+	 *  and saves the 2 seconds failure.
+	 *  See eMMCC spec for partition configuration registers (CSD
+	 *  Byte 179)
+	 */
+	struct mmc *mmc = find_mmc_device(CONFIG_FASTBOOT_FLASH_MMC_DEV);
+	if ( mmc && (EXT_CSD_EXTRACT_PARTITION_ACCESS(mmc->part_config))){
+	    /* no partition table available in the boot partitions */
+	    printf("accessing the boot partition, skipping partition table read\n");
+	    return  -1;
+	}
```

Comment:
So that would only work if someone uses U-Boot in part 1. If for some reason one decides to have U-Boot in the same partition as the OS that would fail right?

Reply:
No, this would be fine whether uboot is there or not.  This function is a generic function to read the partition table, and the check here is for if the user partition or boot partition is enabled.  If boot is enabled, then no attempt should be made to read the partition table.   The partition table can never be in the boot partition, as the boot partition is not accessible after boot and the partition table is too large to fit in the boot partition.   Since the partition table is always in the user partition, this will check if the boot partition is enabled and if it is, return, saving ~2 seconds and a spurious error.